### ISSUE TITLE
Reblog no longer works with GNU Social

### ISSUE THREAD

**Issue by schiessle** [andstatus/andstatus#424]
I noticed that since a few weeks "Reblog" no longer works with AndStatus v30.00. I'm running GNU Social master. 

If I try to Reblog with AndStatus I get this error messages in the server log:

`
2016-10-13 18:31:06 LOG_DEBUG: [io.schiessle.org:18459.afb3aa32 POST /api/statuses/retweet/.json] Could not find a handler for the given path 'api/statuses/retweet/.json'.
2016-10-13 18:31:06 LOG_ERR: [io.schiessle.org:18459.afb3aa32 POST /api/statuses/retweet/.json] PEAR error: Page not found.
2016-10-13 18:31:06 LOG_ERR: [io.schiessle.org:18459.afb3aa32 POST /api/statuses/retweet/.json] #0 Router->map(string) called at [/var/www/gnusocial/index.php:267]
2016-10-13 18:31:06 LOG_ERR: [io.schiessle.org:18459.afb3aa32 POST /api/statuses/retweet/.json] #1 main() called at [/var/www/gnusocial/index.php:320]
`

as you can see, the message ID is missing, the correct end-point would be:

`
/api/statuses/retweet/<message-id>.json
`

Also tested it with a manual CURL request, if I add the message ID it works

**Comment by yvolk** [comment]
Hello @schiessle 
I created a couple of test cases for this (see a commit above) + successfully reblogged a couple of your posts :-) , see https://loadaverage.org/conversation/7978807#notice-8401073

Maybe the problem is with some concrete message/server/android version... ?

And please try v.30.01 (currently in Beta testing channel https://play.google.com/apps/testing/org.andstatus.app )
Please lookup AndStatus logs https://github.com/andstatus/andstatus/issues/225

**Comment by schiessle** [comment]
I now removed all log files, started AndStatus, refreshed the timeline and tried to reblog (this time a different post from a different server). Afterwards this log files were created: https://wolke.schiessle.org/index.php/s/CL88WqmyPorI37J

Maybe you can see there something useful.

Btw, Android version is 6.0.1

I didn't installed the beta yet because I first need to remove the current version and I don't feel really comfortable with it because I don't want to lose all the settings/accounts. Is there a way to do a backup first?

**Comment by schiessle** [comment]
Here another set of logs, again a different server and post and this time with debug mode enabled. So hopefully this contain some more useful information https://wolke.schiessle.org/index.php/s/Bu0msLubPMUBBCa

**Comment by yvolk** [comment]
1. It looks like, for some reason, "server side message ID" is not stored in your local database. "Reblog command doesn't have a check for empty ID, so empty id propagates to the network...

Please try to "favorite" the same "notice" (message). This should fail also with something like this in the log file:
`createFavorite; msgId not found: NNNN"`
where NNNN is internal ID in AndStatus.

2. In order to allow further investigation please create a backup of your AndStatus data: Settings -> Storage ->Backup

**Comment by schiessle** [comment]
The message ID's are stored on the server side. Reblog also works from the Web interface or if I issue the request manually, both would not be possible if the ID wouldn't be stored server side.

If I try to favourite something the message ID is used, see: https://wolke.schiessle.org/index.php/s/Bu0msLubPMUBBCa?path=%2FFavorite

Don't know why I get a "You have already favorited…", but that's a different issue I can investigate server side.

But why does AndStatus has a message ID for "favourite" but not for "reblog"?

**Comment by schiessle** [comment]
I don't know why I tried it earlier, but I also have another GNU Social account at quitter.no. So I tried to reblog and favorite something with this account and it works. Will try to find out what's the difference server side and will come back to you

**Comment by schiessle** [comment]
OK, I found out why the "favorite" function doesn't work with AndStatus but with the web interface. The message ID is different. On the web interface the message ID is 132805 AndStatus tries to favorite message 132865.

So somehow the message IDs are mixed up but I don't know if this happens on the server side and is a API problem or on the client side.

What API call does AndStatus use to get this meta data from a post? If I know the API call I can debug the server further.

**Comment by yvolk** [comment]
@schiessle AndStatus uses several APIs to get a message, including "favorite".
Most often used API for this is "statuses/home_timeline" ...
The easiest way to figure out, where the problem is: to turn logging network level messages to file (in AndStatus). So all input JSONs with data will be logged...

For a list of APIs used please see org.andstatus.app.net.social.ConnectionTwitter#getApiPath1 and specific to GNU Social:
org.andstatus.app.net.social.ConnectionTwitterGnuSocial#getApiPath1

**Comment by schiessle** [comment]
I noticed that "get message info" used the right message ID. So the situation was:

* get message ID: right ID
* favorite: wrong ID
*reblog: no ID

quite confusing. So I decided to remove AndStatus and all data stored on the mobile phone and set it up from scratch. It seems to work now.... I will keep an eye on it.

**Comment by yvolk** [comment]
@schiessle This is why I asked you to create a backup: in order to find out, what went wrong in the local data...
Maybe SQLite (local) database has got corrupted... The problem is that I never had a chance to see such a database. If I had such a database, I would have added some checks to AndStatus to clearly tell a User, what happened...

**Comment by schiessle** [comment]
You are right, forget completely about the backup but the next time something like this happens I will create a backup and give it to you. Promised! :)
